### PR TITLE
#23 bugfix: handle cases where a backslash is being escaped

### PR DESCRIPTION
**Pull request by mikeogezi**

### CODE REVIEW

**Comment by electrikmilk** [comment]
This can be fixed by simply adding a case for escaping the slash, the rest of your changes refactor the logic around escaping characters which is unnecessary to fix this issue.

I appreciate your effort, but when I tested the code below fixed this issue with no other changes required.

```go
switch next(1) {
    // ...
    case '\\':
        collection.WriteRune('\\')
    // ...
}
```

Thanks for pointing out this issue and for submiting your first PR!

**Comment by mikeogezi** [comment]
@electrikmilk I started with this smaller change but when I tested it, the string was ended up being parsed incorrectly. Specifically, with "\\", we had a case where we captured the escaped backslash, but skipped to the next line. The full captured string ended up being "\\\n" (the escaped backslash *and anew line)

**Comment by electrikmilk** [comment]
> @electrikmilk I started with this smaller change but when I tested it, the string was ended up being parsed incorrectly. Specifically, with "\\", we had a case where we captured the escaped backslash, but skipped to the next line and captured stuff there too. The full captured string ended up being "\\\n..." (the escaped backslash _and a new line_ along with more stuff)

Oh, I see now. I was able to get this to happen as well, my bad.

```ruby
@string = "\\\n\\"

nothing()
```

Screenshot 2023-10-31 at 17 54 21

File: parser.go [hunk 1]
```diff
@@ -1029,9 +1029,10 @@ func collectIntegerValue(valueType *tokenType, value *any, until *rune) {
 
 func collectString() string {
 	var collection strings.Builder
+	escaped := false
```
Comment (electrikmilk) [inline parser.go:1032]: Please use this syntax to declare variables per the code style in the contribution guidelines. https://github.com/electrikmilk/cherri/blob/main/CONTRIBUTING.md

```suggestion
	var escaped = false
```

**Comment by electrikmilk** [comment]
Just one requested change, otherwise LGTM. Thanks again for your contribution! 🤘

**Pull request merged**